### DISCUSSION TITLE
Use of Github Actions for new releases

### DISCUSSION THREAD

**Discussion by FlorianWilhelm** [pyscaffold/pyscaffold#343] [Ideas]
Just a random idea but we could use Github Actions to create a new release every time a new tag is pushed. Could also be too dangerous.

**Comment by abravalheri** [comment]
Hi @FlorianWilhelm, I really like this idea, in fact I wanted, after v4 is released to start working in a Github Actions plugin as well (I suppose it is quite popular these days, and to some extent can also be used for CI).

**Comment by FlorianWilhelm** [comment]
Just to test the v3.3 release I created a little Github Action that seems to work quite fine on master. I pushed a v3.3b1 release that we could test and then to the v3.3 tomorrow or when do you prefer @abravalheri ?

**Reply by FlorianWilhelm** [reply]
Hm... for some reason the templates are not packaged... working on it.

**Reply by FlorianWilhelm** [reply]
Okay, v3.3b2 works. 
Something I noticed which might explain that some users have still problems with packaging check-in files that even correctly reside in `src/pkg_name`. You have to `pip install -e .` before running `python setup.py sdist bdist_wheel`, it seems. Makes totally sense, due to `setuptools-scm` but I wasn't really aware of it since I use editable installs all the time anyway.

**Reply by abravalheri** [reply]
Hi @FlorianWilhelm releasing v3.3 tomorrow really works for me :)
I am very happy with the idea of a Christmas gift to the community 😄 

Thank you very much for solving the problems.
Should we document these problems with the packaging? (I wonder if the tox based approach for v4 will have the same problems...)

**Reply by FlorianWilhelm** [reply]
Cool, then I'll push the v3.3 tag tomorrow on 24th and make a little Twitter announcement, nice 🎅 

Yes, I think we should document this, and also in the contribution docs, we would have to so that everyone knows that each tag will be directly pushed to PyPI. I think we can safely do this on the `v4.0.x` branch to avoid larger merge conflicts later and also version 4.0 should be quite around the corner anyway, right? If it wasn't for the interactive feature that is so hard to implement.

**Reply by abravalheri** [reply]
🤞 v4  is around the corner, I will update the check list of the missing tasks on #235.

[update]
To keep track of tasks for v4, I updated the milestone.
Something to have in mind is that the `edit` extension is an alternative to `iteractive` (they serve the same purpose), so if `iteractive` is too complicated, we could go for `edit`...

**Comment by FlorianWilhelm** [comment]
Happy new year @abravalheri! 

I had to disable the run-on-tag-push functionality of the `make-demo` Github Action workflow. It's just not robust enough and also ran on other branches or not at all. The problem is that right now it's really hard to determine the current branch of some action and there are tons of issues and questions about this as `github.ref` or `GITHUB_REF` may point to `refs/heads/branch` or `refs/tags/tag` or in case of PRs to even other things. 

So I just deactivated that functionality and now we have to trigger it manually which is not so bad I guess until Github provides better ways to specify that some action should run for tags only on a certain branch.

**Reply by abravalheri** [reply]
Thank you very much @FlorianWilhelm. Triggering manually should be fine. Worst case scenario, if we do that in every major release it is already a nice thing. If we remember to do it in minor release also, even better.

**Comment by FlorianWilhelm** [comment]
@abravalheri What do you think about this little gimmick in the [demo repo](https://github.com/pyscaffold/dsproject-demo) of the `pyscaffoldext-dsproject` extension. I added the RTD badge using the Github Action. This way users who check out the demo repo can also directly see how a generated Sphinx doc would look like when using PyScaffold. I could also add this to the [pyscaffold-demo](https://github.com/pyscaffold/pyscaffold-demo)-repo.

**Reply by abravalheri** [reply]
```bash
echo -e "[![ReadTheDocs](https://readthedocs.org/projects/dsproject-demo/badge/?version=latest)](https://dsproject-demo.readthedocs.io/)\n\n$(cat README.md)" > README.md
```

This is super cool, and very simple 👏, you are becoming a master of Github Actions :P 

And I see that you don't even need to create a commit with `cpina/github-action-push-to-another-repository`... Handy!
